lib/utils/utilities.js

```diff
@@ -40,6 +40,57 @@ function getSimulatorsFolder() {
     return simulatorsFolderPath;
 }
 
+function getDeviceId() {
+    var findSimCommand = "cordova run --list --emulator | grep ^iPhone | tail -n1";
+
+    logger.info("running:");
+    logger.info("    " + findSimCommand);
+
+    var findSimResult = shelljs.exec(findSimCommand, {silent: true, async: false});
+
+    if (findSimResult.code > 0) {
+        logger.error("Failed to find simulator we deployed to");
+        return undefined;
+    }
+
+    return findSimResult.output;
+}
+
+function getSimulatorId(findSimResult) {
+    var split = findSimResult.split(", ");
+
+    // Format of the output is "iPhone-6s-Plus, 9.1"
+    // Extract the device name and the version number
+    var device = split[0].replace(/-/g, " ").trim();
+    var version = split[1].trim();
+
+    // Next, figure out the ID of the simulator we found
+    var instrCommand = "instruments -s devices | grep ^iPhone";
+    logger.info("running:");
+    logger.info("    " + instrCommand);
+
+    var instrResult = shelljs.exec(instrCommand, {silent: true, async: false});
+
+    if (instrResult.code > 0) {
+        logger.error("Failed to get the list of simulators");
+        return undefined;
+    }
+
+    // This matches <device> (<version>) [<simulator-id>]
+    var simIdRegex = /^([a-zA-Z\d ]+) \(([\d.]+)\) \[([a-zA-Z\d\-]*)\].*$/;
+
+    var simId = null;
+    var lines = instrResult.output.split(/\n/);
+    lines.forEach(function(line) {
+        var simIdMatch = simIdRegex.exec(line);
+        if (simIdMatch && simIdMatch.length === 4 && simIdMatch[1] === device && simIdMatch[2] === version) {
+            simId = encodeURIComponent(simIdMatch[3]);
+        }
+    });
+
+    return simId;
+}
+
 function getSimId() {
```

Comment:
Should `getSimID` be deleted now that you've defined `getSimulatorId` and `getDeviceId`?